doc/reference/reference_lua/pub-sub.rst

```diff
@@ -0,0 +1,172 @@
+Each subscription is defined by the certain key.
+The main feature of the subscriptions is a one-time action.
+It means that if the server generates too many events and a client is too slow,
+the server will not allocate additional memory.
+
+The system processes the following events:
+
+*   ``box.id``
+*   ``box.status``
+*   ``box.election``
+*   ``box.schema``
+
+In response to each event, the server sends back certain IPROTO fields.
+
+Built-in events for pub/sub
+---------------------------
+
+One of the most important purposes of the built-in events is :ref:`master discovery <cfg_basic-master>`.
+It is necessary to know the master node in order to send changes to a correct instance
+or to read the most recent data.
+In addition, there are more built-in events for other mutable properties such as the change of the leader's
+state, its election role and term, the change of the schema version,
+and the state of the instance.
+
+Built-in events have a special naming schema -- theirs names always start with the ``box`` prefix.
+This prefix is reserved for the built-in events. It means that you cannot create new events with it.
+
+Note that built-in events can't be overridden.
+It means that the user cannot call
+:doc:`box.broadcast('box.id', any_data) </reference/reference_lua/box_watchers/broadcast>`.
+
+Below is a table of all events that includes the name of the event and its description.
+
+..  container:: table
+
+    ..  list-table::
+        :widths: 50 50
+
+        *   -   Built-in event
+            -   Description
+
+        *   -   :doc:`box.id </reference/reference_lua/box_info>`
+            -   Identification of the instance. Changes are particularly rare. Some
+                values never change or change only once. For example, the UUID of the instance never
+                changes after the first :doc:`box.cfg </reference/reference_lua/box_cfg>`.
+                However, it is not known before ``box.cfg`` is called.
+                The replicaset UUID is unknown until the instance joins a replicaset or
+                boots a new one, but events are supposed to start working before then --
+                right when the listen launches. The numeric instance ID is known only after
+                registration. For anonymous replicas, the value is ``0`` until they are officially registered.
+
+        *   -   box.status
+            -   Generic blob about the instance status. There are the most frequently used
+                and not frequently changed :doc:`config options </reference/reference_lua/box_cfg>` and
+                :doc:`box.info </reference/reference_lua/box_info>` fields.
+
+        *   -   box.election
+            -   All the required parts of :doc:`box.info.election </reference/reference_lua/box_info/election>`
+                needed to find out who is the most recent writable leader.
+
+        *   -   box.schema
+            -   Schema-related data. Currently, it contains only the version.
```

Comment:
I don't understand when these events are generated. IMO the documentation should have a separate sub-section for each built-in event where you would describe what triggers the event and what fields the event contains. Separating event description from event content description looks confusing.